init_process.go

```diff
@@ -0,0 +1,409 @@
+		if v.Field(i).Interface().(string) == "" {
+			continue
+		}
+		writeContent := "export " + t.Field(i).Name + "=" + v.Field(i).Interface().(string) + "\n"
+		_, err = initProcessEntrypointFile.WriteString(writeContent)
+		if err != nil {
+			fmt.Println(err)
+			log.Fatalf("Cannot write %s in init process entrypoint file\n", writeContent[:len(writeContent)-2])
+		}
+	}
+}
+
+//create a new main config file
+func createMainConfigFile(filePath string) {
+	mainConfigFile, err := os.Create(filePath)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot create main config file")
+	}
+	defer mainConfigFile.Close()
+}
+
+//add @INCLUDE original main config file to the new main config file
+func includeOriginalMainConfigFile(mainConfigFilePath, originalMainConfigFilePath string) {
+	mainConfigFile, err := os.OpenFile(mainConfigFilePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Unable to read main config file: %s\n", mainConfigFilePath)
+	}
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + originalMainConfigFilePath + "\n"
+	_, err = mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+//create fluent bit command to use new main config file
+func createCommand(command *string, filePath string) {
+	*command = *command + " -c /" + filePath
+}
+
+//create a folder to store config files user specified
+func createConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, os.ModePerm)
+}
+
+//get config files user specified according to environment variables
+func getAllConfigFiles() {
+
+	//get all env vars
+	envs := os.Environ()
+	//find all env vars match specified prefix
+	for _, env := range envs {
+		var envKey string
+		var envValue string
+		env_kv := strings.SplitN(env, "=", 2)
+		if len(env_kv) != 2 {
+			log.Printf("Unrecognizable environment variables: %s\n", env)
+			continue
+		}
+
+		envKey = string(env_kv[0])
+		envValue = string(env_kv[1])
+
+		matched_s3, _ := regexp.MatchString("aws_fluent_bit_s3_", envKey)
+		matched_file, _ := regexp.MatchString("aws_fluent_bit_file_", envKey)
+		//if this env var's value is an arn
+		if matched_s3 {
+			getS3ConfigFile(envValue)
+		}
+		//if this env var's value is a path of our built-in config file
+		if matched_file {
+			getBuiltInConfigFile(envValue)
+		}
+	}
+}
+
+//copy built-in config file to config file folder
+func getBuiltInConfigFile(path string) {
+
+	fileName := strings.SplitN(path, "/", -1)
+
+	src, err := os.Open(path[1:]) //remove first "/"
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot find built-in config file: %s\n", path[1:])
+	}
+	defer src.Close()
+
+	dstPath := ConfigFileFolderPath + "/" + fileName[len(fileName)-1]
+	dst, err := os.Create(dstPath)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot creat config file folder")
```

Comment:
why do we need to create a separate folder for built-in configs? We can just use it as it is included in our repo: /ecs/...

Reply:
This actually came from a discussion with Bingfeng, he suggested that we eventually output all the configuration files used for the customer, and I'm still thinking about how to do it here..

The `cat` command in the [init_process_entrypoint.sh] is also for the same purpose

Will design a more reasonable way and modify it.